symbology/create-symbols-from-web-styles/src/main/java/com/esri/samples/create_symbols_from_web_styles/CreateSymbolsFromWebStylesSample.java

```diff
@@ -0,0 +1,316 @@
+
+      // create a feature layer from a service
+      FeatureLayer featureLayer = new FeatureLayer(new ServiceFeatureTable(
+        "http://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/LA_County_Points_of_Interest/FeatureServer/0"));
+
+      // add the feature layer to the map's operational layers
+      map.getOperationalLayers().add(featureLayer);
+
+      // display an error if the feature layer fails to load
+      featureLayer.addDoneLoadingListener(() -> {
+        if (featureLayer.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+          new Alert(Alert.AlertType.ERROR, "Feature layer failed to load. Details: \n"
+            + featureLayer.getLoadError().getCause().getMessage()).show();
+        }
+      });
+
+      // create a unique value renderer
+      UniqueValueRenderer uniqueValueRenderer = new UniqueValueRenderer();
+
+      // add the name of a field from the feature layer data that symbols will be mapped to
+      uniqueValueRenderer.getFieldNames().add("cat2");
+
+      // set the unique value renderer on the feature layer
+      featureLayer.setRenderer(uniqueValueRenderer);
+
+      // create a symbol style from a web style
+      symbolStyle = new SymbolStyle("Esri2DPointSymbolsStyle", null);
+
+      // display an error if the symbol style fails to load
+      symbolStyle.addDoneLoadingListener(() -> {
+        if (symbolStyle.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+          new Alert(Alert.AlertType.ERROR, "Error: could not load symbol style. Details: \n"
+            + symbolStyle.getLoadError().getMessage()).show();
+        }
+      });
+
+      // setup the UI for the legend
+      setupLegend();
+
+      // create a list of the required symbol names from the web style
+      ArrayList<String> symbolNames = new ArrayList<>();
+      symbolNames.addAll(Arrays.asList("atm", "beach", "campground", "city-hall", "hospital", "library", "park",
+        "place-of-worship", "police-station", "post-office", "school", "trail"));
+
+      for (String symbolName : symbolNames) {
+
+        // search for each symbol in the symbol style
+        List<String> searchKey = new ArrayList<>();
+        searchKey.add(symbolName);
+        ListenableFuture<Symbol> searchResult = symbolStyle.getSymbolAsync(searchKey);
+        searchResult.addDoneListener(() -> {
+          try {
+            // get the symbol from the search result
+            Symbol symbol = searchResult.get();
+
+            // get a list of all categories to be mapped to the symbol
+            List<String> categories = mapSymbolNameToField(symbolName);
+
+            for (String category : categories) {
+
+              // create a unique value for each category
+              List<Object> categoryValue = new ArrayList<>();
+              categoryValue.add(category);
+              UniqueValueRenderer.UniqueValue uniqueValue = new UniqueValueRenderer.UniqueValue(
+                "", symbolName, symbol, categoryValue);
```

Comment:
I haven't been able to test this, but we might be able to do the same trick above and simply pass a `Collections.singletonList(item)` into the method call. 
```suggestion
              UniqueValueRenderer.UniqueValue uniqueValue = new UniqueValueRenderer.UniqueValue(
                "", symbolName, symbol, Collections.singletonList(category));
```

It's possibly a bit less easy to read though, so up to you whether you choose to do it this way